tests/test_pipe_outputs_to_inputs.py

```diff
@@ -0,0 +1,105 @@
+# Tests the cases where outputs from prior phases are used as sources to later
+# phases.
+from collections import defaultdict
+import os
+import pytest
+from pathlib import Path
+from phaser import (
+    Column,
+    context_step,
+    IntColumn,
+    Phase,
+    Pipeline,
+    row_step,
+)
+from phaser.io import read_csv, save_csv
+
+@row_step
+def increment_counts(row, context):
+    parent_id = row['parent_id']
+    counts = context.get('counts')
+    counts[parent_id] += 1
+    return row
+
+# TODO: this is an annoying and repetitive reshaping of data that needs to be
```

Comment:
Maybe there are enough sources and outputs that are dictionaries for mapping, rather than rows, that we should just support that as a IO type? 
* Loading in "State capitals" as mapping data to use to enrich - would be able to load it in as a dictionary from the beginning
* Outputting "Employee count by manager" as an extra output would save it out as a dictionary
* and in this example of generating mapping data in one phase and using it in another, it would stay in that format throughout

The format to save mapping data as a dict can still be a CSV with a index column as the first column, or when we support JSON that has a native dict foramt.

Reply:
Let's do some design thinking over in #93. I'll start by copying over this comment of yours.